changelog.d/10962.bugfix

```diff
@@ -0,0 +1 @@
+Fix [MSC2716](https://github.com/matrix-org/matrix-doc/pull/2716) `/batch_send` endpoint rejecting subsequent batches with unknown batch ID error in existing room versions from the room creator.
```

Comment:
Better way to explain?

In existing room versions, the room creator can use the `/batch_send` endpoint. But previously because of the logic flaw, we didn't add the insertion event batch ID's to the database table so when you tried to reference that batch in another chunk, we couldn't find the insertion event and it would throw a `400` `No insertion event corresponds to the given ?batch_id` error.